File: 3/3-1.py
import re
import sys

if len(sys.argv) < 2:
    file = sys.stdin
else:
    file = open(sys.argv[1], 'rt')

total = 0
for line in file:
    for match in re.findall(r'mul\((\d+),(\d+)\)', line):
        total += int(match[0]) * int(match[1])

print(total)

File: 3/3-1.cpp
#include <iostream>
#include <fstream>
#include <string>
#include <regex>

using namespace std;

void main(int argc, char* argv[])
{
    istream &input = argc > 1 ? static_cast<istream&>(ifstream{argv[1]}) : cin;
    regex mul_regex(R"(mul\((\d+),(\d+)\))");

    int total{0};
    string s;
    while (input >> s) {
        for (sregex_iterator i = sregex_iterator(s.begin(), s.end(), mul_regex); i != sregex_iterator(); ++i) {
             smatch match = *i;
             total += stoi(match[1]) * stoi(match[2]);
        }
    }

    cout << total << endl;
}